v2/internal/util/match/glob_matcher.go

```diff
@@ -78,20 +74,21 @@ func (gm *globMatcher) IsRestrictive() bool {
 	return gm.glob != "" && gm.glob != "*"
 }
 
-func (gm *globMatcher) createRegex() {
-	g := regexp.QuoteMeta(gm.glob)
+// String returns the literal we match
+func (gm *globMatcher) String() string {
+	return gm.glob
+}
+
+func newGlobRegex(glob string) (*regexp.Regexp, error) {
```

Comment:
With this construction, it's not possible for the regex to panic; no error return required.

Reply:
It's the construction of the regex string that makes panic impossible. It also makes an error impossible.

`regexp.QuoteMeta` returns a valid regex that matches the exact string given - all regex operators are fully escaped. This regex will compile without panic or error.

The first replacement turns quoted `*` wildcards into matches. This regex will compile without issues.
The second replacement turns quoted `?` wildcards into matches. This regex will also compile without issues.

Reply:
> It also makes an error impossible.

How do we actually know this is true? I looked at [regex parse](https://go.dev/src/regexp/syntax/parse.go) and there seem to be (to me) at least a few error codes which could occur even with the protection we have here.
1. `ErrInvalidUTF8`
2. `ErrLarge`

There might be others as well (I haven't examined the implementation of regexp in detail, nor should we really have to in any case because the impl might change).

IMO the choices here are:
1. Use `MustCompile`, which could panic. AFAIK the only panics would be because of wonky user input, most things will work.
2. Use `Compile` and return the error, which is what I am doing here, primary to avoid the panic and give the caller the chance to attach additional details before deciding if it needs to exit or not.

Are you arguing for 1? Or are you arguing for some other option I don't see?

Reply:
Changed this